realm.go

```diff
@@ -83,9 +86,13 @@ func (l *localClient) onLeave(session ID) {
 }
 
 func (r *Realm) handleSession(sess Session) {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
This looks dangerous. I noticed you have different logic in your fork. I'll probably use that instead.